### PR TITLE
add banking77 dataset

### PR DESCRIPTION
**Pull request by dkajtoch**
Intent classification/detection dataset from banking category with 77 unique intents.

### CODE REVIEW

File: datasets/banking77/README.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+{
+  'label': 0, # integer label corresponding to "card_arrival" intent
+  'text': 'I am still waiting on my card?'
+}
+```
+
+### Data Fields
+
+- `text`: a string feature.
+- `label`: One of classification labels (0-76) corresponding to unique intents.
```
Comment (lhoestq) [inline datasets/banking77/README.md:96]: Could you add the list of the label names please ?
Ideally users should be able to tell which label name correspond to which label id (for example, 0 = card_arrival)

File: datasets/banking77/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+| ---            |   --- |
+| Train examples | 10 003 |
+| Test examples | 3 080 |
+| Number of intents | 77 |
+
+## Dataset Creation
+
+### Curation Rationale
+
+[N/A]
```
Comment (lhoestq) [inline datasets/banking77/README.md:110]: Can you use `[More Information Needed]` instead please ?
This also applies for the other mentions of `N/A`

File: datasets/banking77/banking77.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                text, label = row
+                yield id_, {"text": text, "label": label}
+
+
+if __name__ == "__main__":
+    from datasets import load_dataset
+
+    data = load_dataset("./banking77.py")
+    print(data["train"][0])
+    # print(data["train"].features["label"])
```
Comment (lhoestq) [inline datasets/banking77/banking77.py:178]:
```suggestion
```

**Review by lhoestq — COMMENTED** [review]
Cool thank you for adding this one !
This looks all good, good job :)

I just added minor comments:

File: datasets/banking77/README.md (hunk 1 above)
Comment (dkajtoch) [reply datasets/banking77/README.md:96]: Sure! just have a question. I used `ClassLabel` for text labels and it seems to encode names with integers. Do you want me to make a list which label correspond to which category (I assume ClassLabel goes in order of names)?

File: datasets/banking77/README.md (hunk 2 above)
Comment (dkajtoch) [reply datasets/banking77/README.md:110]: I will try to be more informative

**Comment by dkajtoch** [comment]
@lhoestq I updated files

File: datasets/banking77/README.md (hunk 1 above)
Comment (lhoestq) [reply datasets/banking77/README.md:96]: What you did is perfect thanks :)

File: datasets/banking77/README.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+English
+
+## Dataset Structure
+
+### Data Instances
+
+An example of 'train' looks as follows:
+```
+{
+  'label': 0, # integer label corresponding to "card_arrival" intent
```
Comment (lhoestq) [inline datasets/banking77/README.md:88]:
```suggestion
  'label': 11, # integer label corresponding to "card_arrival" intent
```

**Review by lhoestq — APPROVED** [review]
Thanks LGTM :)

**Pull request merged**